apps/aefate/src/aefa_chain_api.erl

```diff
@@ -608,34 +724,50 @@ aens_resolve_from_pstate(NameString, Key, PState) ->
     end.
 
 aens_preclaim(Pubkey, Hash, #state{} = S, VmVersion) when ?IS_ONCHAIN(S) ->
-    case VmVersion >= ?VM_FATE_SOPHIA_2 of
-        true ->
-            PreclaimTTL = aec_governance:name_preclaim_expiration(),
-            eval_primops([aeprimop:name_preclaim_op(Pubkey, Hash, PreclaimTTL)], S);
-        false ->
-            eval_primops([aeprimop:name_preclaim_op(Pubkey, Hash, 0)], S)
-    end.
+    PreclaimTTL =
+        case VmVersion >= ?VM_FATE_SOPHIA_2 of
+            true ->
+                aec_governance:name_preclaim_expiration();
+            false ->
+                0
+        end,
+    eval_primops([ aeprimop:name_preclaim_op(Pubkey, Hash, PreclaimTTL)
+                 , tx_event_op(aens_preclaim, {Pubkey, Hash, ttl(PreclaimTTL, S)})
+                 ], S).
 
 aens_claim(Pubkey, NameBin, SaltInt, NameFee, #state{} = S) when ?IS_ONCHAIN(S) ->
     PreclaimDelta = aec_governance:name_claim_preclaim_delta(),
-    Instructions = [ aeprimop:name_claim_op(Pubkey, NameBin, SaltInt, NameFee, PreclaimDelta) ],
+    Instructions = [ aeprimop:name_claim_op(Pubkey, NameBin, SaltInt, NameFee, PreclaimDelta)
+                   , tx_event_op(aens_claim, {Pubkey, NameBin, SaltInt, NameFee,
+                                              ttl(PreclaimDelta, S)})],
     eval_primops(Instructions, S, size_gas([NameBin])).
 
 aens_transfer(FromPubkey, HashBin, ToPubkey, #state{} = S) when ?IS_ONCHAIN(S) ->
-    Instructions = [aeprimop:name_transfer_op(FromPubkey, account, ToPubkey, HashBin)
+    %% We need to lookup the Name for the tx_event op, unfortunately
```

Comment:
Would it be possible to keep the Name until this part in the call-chain or would that be too ugly? Looking up the name could be costly, or maybe not 🤔

Reply:
It turned out that we didn't need the lookup - nor even the name string - at all.